example/game/src/main/java/no/nordicsemi/android/ble/example/game/server/viewmodel/ServerViewModel.kt

```diff
@@ -0,0 +1,170 @@
+                val showDone = index == questions.questions.size - 1
+
+                QuestionState(
+                    question = question,
+                    questionIndex = index,
+                    totalQuestions = questions.questions.size,
+                    showDone = showDone
+                )
+            }
+
+            // TODO send questions
+            clients.forEach {
+                if (gameStartedFlag.value){
+//                    it.sayHello()
+//                    it.gameStart(gameStartedFlag.value)
+                    it.sendQuestion(questions.questions[0])
+                }
+            }
+            // start game :)
+            _uiState.emit(Questions(questionStates))
+            _state.emit(Round(questions.questions[0]))
+        }
+    }
+
+    private fun startServer(){
+        Log.d("AAA", "startServer")
+        advertiser.startAdvertising()
+        serverManager.setServerObserver(object : ServerObserver {
+
+            override fun onServerReady() {
+                Log.w("AAA", "onServerReady")
+            }
+
+            override fun onDeviceConnectedToServer(device: BluetoothDevice) {
+                Log.w("AAA", "onDeviceConnectedToServer")
+
+                ServerConnection(getApplication(), viewModelScope, device)
+                    .apply {
+                        stateAsFlow()
+                            .onEach { connectionState ->
+                                val currentState = _state.value
+                                when (connectionState) {
+                                    ConnectionState.Ready -> {
+                                        clients.add(this)
+                                        _state.value = WaitingForPlayers(clients.size)
+                                    }
+                                    is ConnectionState.Disconnected -> {
+                                        clients.remove(this)
+
+                                        when (currentState) {
+                                            is WaitingForPlayers -> {
+                                                _state.value = WaitingForPlayers(clients.size)
+                                            }
+                                            else -> {
+                                                // TODO handle disconnection during game
+                                            }
+                                        }
+                                    }
+                                    else -> {}
+                                }
+                            }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        replies
+                            .onEach { /* TODO handle response received */ }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
```

Comment:
I have feeling that those apply are beeing overused. As a result the code moves to much to the right and reading is harder than easier.